x/concentrated-liquidity/incentives.go

```diff
@@ -949,46 +961,68 @@ func (k Keeper) claimAllIncentivesForPosition(ctx sdk.Context, positionId uint64
 			// If the claimed incentives are forfeited, deposit them back into the accumulator to be distributed
 			// to other qualifying positions.
 			if positionAge < supportedUptimes[uptimeIndex] {
+				forfeitedIncentivesForUptime := collectedIncentivesForUptime
+
 				totalSharesAccum, err := uptimeAccum.GetTotalShares()
 				if err != nil {
 					return sdk.Coins{}, sdk.DecCoins{}, err
 				}
 
 				if totalSharesAccum.IsZero() {
-					pool, err := k.getPoolById(ctx, position.PoolId)
-					if err != nil {
-						return sdk.Coins{}, sdk.DecCoins{}, err
-					}
-
 					// If totalSharesAccum is zero, then there are no other qualifying positions to distribute the forfeited
 					// incentives to. This might happen if this is the last position in the pool and it is being withdrawn.
 					// Therefore, we send the forfeited amount to the community pool in this case.
-					err = k.communityPoolKeeper.FundCommunityPool(ctx, collectedIncentivesForUptime, pool.GetIncentivesAddress())
+					err = k.communityPoolKeeper.FundCommunityPool(ctx, forfeitedIncentivesForUptime, pool.GetIncentivesAddress())
 					if err != nil {
 						return sdk.Coins{}, sdk.DecCoins{}, err
 					}
 
-					forfeitedIncentivesForPosition = forfeitedIncentivesForPosition.Add(sdk.NewDecCoinsFromCoins(collectedIncentivesForUptime...)...)
+					forfeitedIncentivesForPosition = forfeitedIncentivesForPosition.Add(sdk.NewDecCoinsFromCoins(forfeitedIncentivesForUptime...)...)
 					continue
 				}
 
 				var forfeitedIncentivesPerShare sdk.DecCoins
-				for _, coin := range collectedIncentivesForUptime {
-					// updated forfeitedIncentivesPerShare to add back = collectedIncentivesPerShare / totalSharesAccum
-					forfeitedIncentivesPerShare = append(forfeitedIncentivesPerShare, sdk.NewDecCoinFromDec(coin.Denom, coin.Amount.ToDec().Add(dust.AmountOf(coin.Denom)).Quo(totalSharesAccum)))
+				for _, forfeitedCoin := range forfeitedIncentivesForUptime {
+					// Total forfeited amount includes the dust since it gets added back into the accumulator to be distributed to other LPs.
+					forfeitedDecCoin := sdk.NewDecCoinFromDec(forfeitedCoin.Denom, forfeitedCoin.Amount.ToDec().Add(dust.AmountOf(forfeitedCoin.Denom)))
+
+					// To determine the amount of liquidity we must distribute the forfeited incentives to, we add the current tick liquidity as well as the qualifying balancer shares.
+					qualifyingLiquidity := pool.GetLiquidity().Add(qualifyingBalancerShares)
```

Comment:
Note to reviewer: This is primary issue 2, we need to use the pool's current tick liquidity in combination with the qualifying balancer shares. We were previously using the accumulator's total share accum value which isn't needed here at all.